google-cloud-bigquery/acceptance/bigquery/advanced_test.rb

```diff
@@ -0,0 +1,167 @@
+          spells.bytes "icon", mode: :nullable
+          spells.timestamp "last_used", mode: :nullable
+        end
+        schema.time "tea_time", mode: :nullable
+        schema.date "next_vacation", mode: :nullable
+        schema.datetime "favorite_time", mode: :nullable
+      end
+      t.insert example_table_rows
+    end
+    t
+  end
+
+  def example_table_rows
+    [
+      { id: 1,
+        name: "Bilbo",
+        age: 111,
+        weight: 67.2,
+        is_magic: false,
+        scores: [],
+        spells: [],
+        tea_time: Google::Cloud::Bigquery::Time.new("10:00:00"),
+        next_vacation: Date.parse("2017-09-22"),
+        favorite_time: Time.parse("2031-04-01T05:09:27").utc.to_datetime
+      }, {
+        id: 2,
+        name: "Gandalf",
+        age: 1000,
+        weight: 198.6,
+        is_magic: true,
+        scores: [100.0, 99.0, 0.001],
+        spells: [
+          { name: "Skydragon",
+            discovered_by: "Firebreather",
+            properties: [
+              { name: "Flying", power: 1.0 },
+              { name: "Creature", power: 1.0 },
+              { name: "Explodey", power: 11.0 }
+            ],
+            icon: File.open("acceptance/data/kitten-test-data.json", "rb"),
+            last_used: Time.parse("2015-10-31 23:59:56 UTC")
+          }
+        ],
+        tea_time: Google::Cloud::Bigquery::Time.new("15:00:00"),
+        next_vacation: Date.parse("2666-06-06"),
+        favorite_time: Time.parse("2001-12-19T23:59:59").utc.to_datetime
+      }, {
+        id: 3,
+        name: "Sabrina",
+        age: 17,
+        weight: 128.3,
+        is_magic: true,
+        scores: [],
+        spells: [
+          { name: "Talking cats",
+            icon: StringIO.new("iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAYAAACqaXHeAAAAAXNSR0IArs4c6QAAABxpRE9UAAAAAgAAAAAAAAAgAAAAKAAAACAAAAAgAAABxj2CfowAAAGSSURBVHgB7Jc9TsNAEIX3JDkCPUV6KlpKFHEGlD4nyA04ACUXQKTgCEipUnKGNEbP0otentayicZ24SlWs7tjO/N9u/5J2b2+NUtuZcnwYE8BuQPyGZAPwXwLLPk5kG+BJa9+fgfkh1B+CeancL4F8i2Q/wWm/S/w+XFoTseftn0dvhu0OXfhpM+AGvzcEiYVAFisPqE9zrETJhHAlXfg2lglMK9z0f3RBfB+ZyRUV3x+erzsEIjjOBqc1xtNAIrvguybV3A9lkVHxlEE6GrrPb/ZvAySwlUnfCmlPQ+R8JCExvGtcRQBLFwj4FGkznX1VYDKPG/f2/MjwCksXACgdNUxJjwK9xwl4JihOwTFR0kIF+CABEPRnvsvPFctMoYKqAFSAFaMwB4pp3Y+bodIYL9WmIAaIOHxo7W8wiHvAjTvhUeNwwSgeAeAABbqOewC5hBdwFD4+9+7puzXV9fS6/b1wwT4tsaYAhwOOQdUQch5vgZCeAhAv3ZM31yYAAUgvApQQQ6n5w6FB/RVe1jdJOAPAAD//1eMQwoAAAGQSURBVO1UMU4DQQy8X9AgWopIUINEkS4VlJQo4gvwAV7AD3gEH4iSgidESpWSXyyZExP5lr0c7K5PsXBhec/2+jzjuWtent9CLdtu1mG5+gjz+WNr7IsY7eH+tvO+xfuqk4vz7CH91edFaF5v9nb6dBKm13edvrL+0Lk5lMzJkQDeJSkkgHF6mR8CHwMHCQR/NAQQGD0BAlwK4FCefQiefq+A2Vn29tG7igLAfmwcnJu/nJy3BMQkMN9HEPr8AL3bfBv7Bp+7/SoExMDjZwKEJwmyhnnmQIQEBIlz2x0iKoAvJkAC6TsTIH6MqRrEWUMSZF2zAwqT4Eu/e6pzFAIkmNSZ4OFT+VYBIIF//UqbJwnF/4DU0GwOn8r/JQYCpPGufEfJuZiA37ycQw/5uFeqPq4pfR6FADmkBCXjfWdZj3NfXW58dAJyB9W65wRoMWulryvAyqa05nQFaDFrpa8rwMqmtOZ0BWgxa6WvK8DKprTmdAVoMWulryvAyqa05nQFaDFrpa8rwMqmtOb89wr4AtQ4aPoL6yVpAAAAAElFTkSuQmCC"),
```

Comment:
I'd like to keep it in Ruby so I can control the values. This structure comes from Omar's integration test issue